packages/@aws-cdk/aws-secretsmanager/test/secret.test.ts

```diff
@@ -687,6 +687,17 @@ test('fromSecretCompleteArn - grants', () => {
   });
 });
 
+test('fromSecretCompleteArn - can be assigned to a property with type number', () => {
+  const secretArn = 'arn:aws:secretsmanager:eu-west-1:111111111111:secret:MySecret-f3gDy9';
+  const secret = secretsmanager.Secret.fromSecretCompleteArn(stack, 'Secret', secretArn);
+  new lambda.Function(stack, 'MyFunction', {
+    code: lambda.Code.fromInline('foo'),
+    handler: 'bar',
+    runtime: lambda.Runtime.NODEJS,
+    memorySize: cdk.Token.asNumber(secret.secretValueFromJson('LambdaFunctionMemorySize')),
+  });
+});
```

Comment:
You need to synthesize the Stack for the error to be seen 🙂. Adding any assertion, like `expect(stack).to(haveResourceLike('AWS::Lambda::Function')`, or using `SynthUtils.synthesize()`, will do the trick 🙂.

Reply:
Oh makes sense, of course! Will update :)